app/models/restaurant.rb

```diff
@@ -20,4 +42,26 @@ class Restaurant < ApplicationRecord
       message: 'must be a URL for GIF, JPG or PNG image.'
   }
 
+
+  def get_rating
+    num_of_reviews = 0
+    total_rating = 0
+
+    if reviews.present?
+      reviews.each do |review|
+        total_rating += review.rating
+        num_of_reviews += 1
+      end
+      total_rating / num_of_reviews
+    else
+      0
+    end
+
+
+
+  end
```

Comment:
I think you could have replaced this entire thing with `reviews.average(:rating)`.
(Provided by https://apidock.com/rails/ActiveRecord/Calculations/average)

Alternatively:
```ruby
reviews.map { |r| r.rating }.average

# or in shorthand:
reviews.map(&:rating).average
```
I think that is provided by [ActiveSupport](http://guides.rubyonrails.org/active_support_core_extensions.html).

You can also use `sum` and divide by `reviews.size`.
https://stackoverflow.com/questions/1341271/how-do-i-create-an-average-from-a-ruby-array

Reply:
There are a lot of things you can do much more easily in Ruby and even more so in Rails.
It's worth searching for them.

Reply:
Nice!! thanks!!!
changed to: 
reviews.size == 0 ? "-" : reviews.average(:rating).round